shm_wrapper/shm_wrapper.c

```diff
@@ -837,12 +859,18 @@ int log_data_write(char* key, param_type_t type, param_val_t value) {
         }
     }
 
-    // return if we didn't find the log data corresponding to this key
+    // return if we ran out of keys for log data
     if (idx == UCHAR_MAX) {
         log_printf(ERROR, "Maximum number of %d log data keys reached. can't add key %s", UCHAR_MAX, key);
         return -1;
     }
 
+    // 64 is max key size in log data SHM block
+    if (strlen(key) >= 64) {
+        log_printf(ERROR, "Key name %s for log data is longer than 64 characters", key);
+        return -2;
+    }
+
```

Comment:
Can you explain what a log data key is? I'm aware that the inputs protobuf accounts for 64 different buttons available across the gamepad and keyboard. Is this related to that, such as a student somehow exceeding that limit?

Reply:
This is for a different function, the `Robot.log` to log custom data to Dawn. They `key` here is something the students choose to describe their data. I just added this cus I realized my previous implementation didn't correctly check for an error. Its not related to the main PR on inputs at all.